content/HTTPS is not enough - Pablo Blanco.md

```diff
@@ -0,0 +1,218 @@
+and a client. This is typically used to secure communications between web
+servers and a browser, or a mail server and a mail client.
+
+**What is HTTPS?**
+
+HTTPS (Hypertext Transfer Protocol Secure) is a secure version of the HTTP
+protocol that uses the SSL/TLS protocol for encryption and authentication.
+
+## Why only HTTPS is not enough?
+
+Encrypted communications are an essential requirement for health, banks,
+financial and other applications where information security is crucial.
+
+To improve accessibility, most secure websites run an HTTP(301) service to
+redirect users from HTTP to HTTPS endpoints. That redirection **does not** fully
+protect to the user since it can be intercepted between when the visitor
+requests the HTTP version of the site, and when it reaches the HTTPS
+destination, this gives the attacker the opportunity of implement **"man in the
+middle SSLstrip"** and **"session hijacking"** attacks.
+
+Let’s suppose this scenario, an user of our application log into a free WiFi
+access point at an airport and start surfing the web, visiting our online
+banking service to make some transactions, and pay a couple of bills.
+Unfortunately, that access point can be sniffed by an attacker, and it’s really
+easy to intercept all the original HTTP requests and redirect them to a clone of
+the bank's site instead of the real thing. Then, the user’s data is exposed to
+the attacker.
+
+Well maybe you are thinking, it is not supposed that anyone access to its bank
+account on a public Wifi network, and you are completely right, but the same
+scenario can be considered if someone hacks into our **private Wifi network**
+too.
+
+This kind of man in the middle attack is basically performed by the following
+steps:
+
+![White or black hat?](images/https-not-enough-mitm.png)
+
+1.  User asks for `https://somedomain.com`
+2.  MITM intercept that, and ask for `https://somedomain.com` to the Server
+3.  The server responds to MITM `https://somedomain.com`
+4.  MITM responds to the User with `http://somedomain.com`
+
+After that, the attacker can see plain data over the requests made by the user.
+
+## How can we mitigate HTTPS vulnerabilities?
+
+Whenever a browser requests a page from a web server, the server answers to the
+browser with the content along with some useful information, contained in
+**headers**. Many SSL-related vulnerabilities can be mitigated with a special
+type of headers, called security headers that can tell browsers how to treat
+content on a web page such as “whitelist certain types of content from trusted
+sites” or “only load this site over HTTPS”, for this particular vulnerabilities
+there are 2 types of headers we can use, **HSTS** and **CSP**.
+
+### HTTP Strict Transport Security (HSTS)
+
+The HTTP Strict-Transport-Security response header, lets a website tell browsers
+that it should only be accessed using HTTPS strictly, instead of using HTTP and
```

Comment:
it should only be accessed using HTTPS strictly, instead of using HTTP `,` and...